hassio/src/update-available/update-available-card.ts

```diff
@@ -0,0 +1,381 @@
+                      version:
+                        this._addonInfo?.version ||
+                        this.supervisor[this._updateEntry]?.version,
+                      newest_version:
+                        this._addonInfo?.version_latest ||
+                        this.supervisor[this._updateEntry]?.version_latest,
+                    })}
+                  </p>
+                </div>
+                ${!["os", "supervisor"].includes(this._updateEntry)
+                  ? html`
+                      <ha-formfield
+                        .label=${this.supervisor.localize(
+                          "update_available.create_backup"
+                        )}
+                      >
+                        <ha-checkbox
+                          .checked=${this._createBackup}
+                          @click=${this._toggleBackup}
+                        >
+                        </ha-checkbox>
+                      </ha-formfield>
+                    `
+                  : ""}
+              `
+            : html`<ha-circular-progress alt="Updating" size="large" active>
+                </ha-circular-progress>
+                <p class="progress-text">
+                  ${this._action === "update"
+                    ? this.supervisor.localize("update_available.updating", {
+                        name,
+                        version:
+                          this._addonInfo?.version_latest ||
+                          this.supervisor[this._updateEntry]?.version_latest,
+                      })
+                    : this.supervisor.localize(
+                        "update_available.creating_backup",
+                        { name }
+                      )}
+                </p>`}
+        </div>
+        ${this._action === null
+          ? html`
+              <div class="card-actions">
+                ${changelog
+                  ? html`<a .href=${changelog} target="_blank" rel="noreferrer">
+                      <mwc-button
+                        .label=${this.supervisor.localize(
+                          "update_available.open_release_notes"
+                        )}
+                      >
+                      </mwc-button>
+                    </a>`
+                  : ""}
+                <span></span>
+                <ha-progress-button
+                  .disabled=${this._error !== undefined ||
+                  (this._createBackup &&
+                    this.supervisor.info.state !== "running")}
+                  @click=${this._update}
+                  raised
+                >
+                  ${this.supervisor.localize("common.update")}
+                </ha-progress-button>
+              </div>
+            `
+          : ""}
+      </ha-card>
+    `;
+  }
+
+  protected firstUpdated(changedProps: PropertyValues) {
+    super.firstUpdated(changedProps);
+    this._updateEntry =
+      this.addonSlug || this.route.path.substring(1, this.route.path.length);
```

Comment:
It's a bit confusing that this is types, except when it's an add-on, then it's the addon slug. It would be easier to follow if we rename it to `updateType` and set it to `addon` if it's an addon. We know that we can always use `this.addonSlug` to get the slug.